src/vmq_queue.erl

```diff
@@ -516,6 +541,15 @@ handle_session_down(SessionPid, StateName,
             gen_fsm:send_event(self(), drain_start),
             _ = vmq_plugin:all(on_client_offline, [SId]),
             {next_state, state_change({'DOWN', migrate}, wait_for_offline, drain), NewState};
+        {0, wait_for_offline, {cleanup, From}} ->
+            %% Forcefully cleaned up, we have to cleanup remaining offline messages
+            %% we don't cleanup subscriptions!
+            #state{offline=#queue{queue=Q}} = NewState,
+            cleanup_queue(SId, Q),
+            _ = vmq_metrics:incr_queue_unhandled(queue:len(Q)),
+            gen_fsm:reply(From, ok),
+            _ = vmq_plugin:all(on_client_gone, [SId]),
```

Comment:
`on_client_gone` means the client is gone and all it's data with it.. as opposed to `on_client_offline` which is used in case we still have some client data.